cognite/client/exceptions.py

```diff
@@ -211,6 +218,45 @@ class CogniteAPIKeyError(CogniteAuthError):
     """
 
 
+class CogniteAssetHierarchyError(CogniteException, AssertionError):
+    """Cognite Asset Hierarchy validation Error.
+
+    Raised if the given assets form an invalid hierarchy (by CDF standards).
+
+    Note:
+        For historical reasons, we make the error catchable as an AssertionError.
+
+    Args:
+        message (str): TODO
+        hierarchy (AssetHierarchyValidator): An instance of AssetHierarchyValidator that holds various groups
+            of assets that failed validation.
+    """
+
+    def __init__(self, message: str, hierarchy: AssetHierarchyValidator) -> None:
+        self.message = message
+        self._hierarchy = hierarchy
+
+    @property
+    def orphans(self) -> AssetList:
+        return self._hierarchy.orphans
```

Comment:
Note to self: Consider implementing `__getattr__` to give direct access to `self.hierarchy` 🤔